transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -348,21 +499,18 @@ private String generateName(ChannelHandler handler) {
             cache.put(handlerType, name);
         }
 
-        synchronized (this) {
-            // It's not very likely for a user to put more than one handler of the same type, but make sure to avoid
-            // any name conflicts.  Note that we don't cache the names generated here.
-            if (context0(name) != null) {
-                String baseName = name.substring(0, name.length() - 1); // Strip the trailing '0'.
-                for (int i = 1;; i ++) {
-                    String newName = baseName + i;
-                    if (context0(newName) == null) {
-                        name = newName;
-                        break;
-                    }
+        // It's not very likely for a user to put more than one handler of the same type, but make sure to avoid
+        // any name conflicts.  Note that we don't cache the names generated here.
+        if (context0(name) != null) {
+            String baseName = name.substring(0, name.length() - 1); // Strip the trailing '0'.
+            for (int i = 1;; i ++) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) This loop's stop condition tests "" but the incrementer updates "i". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1994)